### DISCUSSION TITLE
Support for non-English (non-Lating) keyboard input sources

### DISCUSSION THREAD

**Discussion by thezvff** [JetBrains/ideavim#950] [Q&A] [answered]
Problem statement: I use 2 language input sources on my machine. Sometimes I need to edit text, which is not English (e.g. Russian). When I want to use vim motions while Russian in selected as lang input source, ideavim bindings don't work.

In Neovim, it can be solved using something like https://github.com/Wansmer/langmapper.nvim. Is there a way I can use vim motions without switching my keyboard input source to English every time?

**Comment by lippfi** [comment, marked as answer]
I can see two ways, actually:
1. The good old `map й q` 
It is possible to write a small script to do all the necessary mappings according to your layout. Here is the code:
```
let russianLayout = 'йцукенгшщзхъфывапролджэячсмитьбю.'
let colemakDHLayout = 'qwfpbjluy;[]arstgmneio'xcdvzkh,./'
let i = 0
while i < len(russianLayout)
    let fromKey = russianLayout[i]
    let toKey = colemakDHLayout[i]
    execute 'map ' . fromKey . ' ' . toKey
    let i += 1
endwhile
```

2. There is https://plugins.jetbrains.com/plugin/9615-ideavimextension

P.S. The script I wrote should work, but you should be cautious when writing your own scripts—as not all Vim Script language functions are supported in IdeaVim
